src/data/api.ts

```diff
@@ -0,0 +1,78 @@
+import { camelCaseObject, getConfig } from '@edx/frontend-platform';
+import { getAuthenticatedHttpClient } from '@edx/frontend-platform/auth';
+
+const getStudioBaseUrl = () => getConfig().STUDIO_BASE_URL as string;
+
+export const getApiWaffleFlagsUrl = (courseId?: string): string => {
+  const baseUrl = getStudioBaseUrl();
+  const apiPath = '/api/contentstore/v1/course_waffle_flags';
+
+  return courseId ? `${baseUrl}${apiPath}/${courseId}` : `${baseUrl}${apiPath}`;
+};
+
+function normalizeCourseDetail(data) {
+  return {
+    id: data.course_id,
+    ...camelCaseObject(data),
+  };
+}
+
+export async function getCourseDetail(courseId: string, username: string) {
+  const { data } = await getAuthenticatedHttpClient()
+    .get(`${getConfig().LMS_BASE_URL}/api/courses/v1/courses/${courseId}?username=${username}`);
+
+  return normalizeCourseDetail(data);
+}
+
+export type WaffleFlagName = (
+  | 'enableCourseOptimizer'
+  | 'useNewAdvancedSettingsPage'
+  | 'useNewCertificatesPage'
+  | 'useNewCourseOutlinePage'
+  | 'useNewCourseTeamPage'
+  | 'useNewCustomPages'
+  | 'useNewExportPage'
+  | 'useNewFilesUploadsPage'
+  | 'useNewGradingPage'
+  | 'useNewGroupConfigurationsPage'
+  | 'useNewHomePage'
+  | 'useNewImportPage'
+  | 'useNewScheduleDetailsPage'
+  | 'useNewTextbooksPage'
+  | 'useNewUnitPage'
+  | 'useNewUpdatesPage'
+  | 'useNewVideoUploadsPage'
+  | 'useReactMarkdownEditor'
+  | 'useVideoGalleryFlow'
+);
+
+/** The default values of waffle flags. helpful if you're overriding them for a test case. */
```

Comment:
Can you say more about these? They're not actually edx-platform defaults; rather, they're the default that we test with, right? Or do these defaults actually have significance on the app?

(I see you pulled these from an existing object in src/data/slice.js, so I'm not trying to nitpick this too much, just looking for clarity).

Reply:
Oh interesting. I assumed that these defaults matched the platform defaults. These are what the MFE will assume while it's waiting to load the actual values from the server. (If we don't want to assume anything, components that depend on waffle flags should block rendering until `isLoading` is false, but I think that's a worse experience.)

I would suggest that we follow this up with a PR to either change these defaults to match the platform defaults, or change the platform defaults to match these. But I think that should be isolated to a separate change, since this PR is trying to be an internal refactor that doesn't change behavior meaningfully.

Reply:
Yeah, I don't think you should fix these in this PR. Could you just amend your comment to say something like "frontend defaults if we fail to load flags from edx-platform -- may drift from edx-platform's actual defaults".

If you feel moved to, you could write an issue and link it here, but I'm not losing sleep over it. FWIW, my inclination here is actually "turn everything on by default in order to maximize test coverage" rather than "try to keep lockstep with edx-platform."

Reply:
I created a GitHub issue https://github.com/openedx/frontend-app-authoring/issues/2094 and put it into the comments.